internal/isoeditor/rhcos.go

```diff
@@ -248,10 +256,14 @@ func (e *rhcosEditor) formatRootfsServiceConfigFile(clusterProxyInfo *ClusterPro
 }
 
 func addFileToArchive(w *cpio.Writer, path string, content string, mode cpio.FileMode) error {
-	// add all the directories in path
+	// add all the directories in path in the correct order, using dirsStack as FILO
+	dirsStack := []string{}
 	for dir := filepath.Dir(path); dir != "" && dir != "/"; dir = filepath.Dir(dir) {
+		dirsStack = append(dirsStack, dir)
+	}
+	for i := len(dirsStack) - 1; i >= 0; i-- {
```

Comment:
The example in go-cpio repo indeed doesn't seem to take into consideration dir structure: https://github.com/cavaliercoder/go-cpio/blob/master/example_test.go
Maybe we could use some dir traversing library for dfs walk on structure: https://github.com/karrick/godirwalk
But seems simple enough for now.

Reply:
i think that  https://github.com/karrick/godirwalk is inspecting real filesystem structure. In our case we just need the parsing of the path as a string.

Reply:
Why was this changed? Seems equivalent to me:
```
09:16:26:~/Documents/scratch/golang/filepath$ cat main.go 
package main

import (
	"fmt"
	"path/filepath"
)

func main() {
	path := "/a/b/c/d/e/f"
	fmt.Println("Old implementation")
	oldDirs(path)
	fmt.Println("New implementation")
	newDirs(path)
}

func oldDirs(path string) {
	for dir := filepath.Dir(path); dir != "" && dir != "/"; dir = filepath.Dir(dir) {
		fmt.Println(dir)
	}
}

func newDirs(path string) {
	dirsStack := []string{}
	for dir := filepath.Dir(path); dir != "" && dir != "/"; dir = filepath.Dir(dir) {
		dirsStack = append(dirsStack, dir)
	}

	for i := len(dirsStack) - 1; i >= 0; i-- {
		fmt.Println(dirsStack[i])
	}
}
09:17:13:~/Documents/scratch/golang/filepath$ go run main.go 
Old implementation
/a/b/c/d/e
/a/b/c/d
/a/b/c
/a/b
/a
New implementation
/a
/a/b
/a/b/c
/a/b/c/d
/a/b/c/d/e
```

Does the order matter?

Reply:
apparently yes, that's why i changed it.
the issue was that we were adding 6 files into the the directory /etc/assisted/network,  and i noticed that the first file was consistently not set in the initramfs. I think what happens when  adding a file /etc/assisted/network/907908.nmconnection is as following:
we add /etc/assisted/network directory, then etc/assisted directory, then /etc, and the we write file /etc/assisted/network/907908.nmconnection. 
since /etc/assisted does not exists, /etc/assisted/network is ignored, but  /etc/assisted directory is taken, since /etc exists.
/etc/assisted/network/907908.nmconnection is also ignored, since /etc/assisted/network is ignored
when adding next file /etc/assisted/network/49085708.nmconnection, /etc/assisted/network is taken , since  /etc/assisted  already exists
that's my explanation, but i am not 100% of it.  but when adding directories in the correct order, we see all the files added

Reply:
I figured there was a reason, I just really want it recorded somewhere.
This is probably some detail of the cpio format that we don't understand fully.

It also makes me wonder if it has something to do with adding the same directory multiple times.
As you said, we're adding them for each file in the same directory.

Reply:
I remember testing that adding a directory that already existed didn't cause a problem (and may in-fact have been necessary), but I don't remember testing multiple files in the same directory.

Reply:
cpio is just an archive format,  kernel knows to read it an build a Fs from it. So i think the order does matter

Reply:
> cpio is just an archive format

Yes, and we clearly don't fully understand it if we don't know if order matters or not :smile: